Archive/Draft/RFCxxxx-PS7-Bootstrapper.md

```diff
@@ -0,0 +1,256 @@
+
+## Shipping in Windows
+
+There are several issues that prevent simply shipping PS7 in Windows:
+
+- Size
+- Support lifecycle differences
+
+### Size constraints
+
+A complete installation of PowerShell 7 currently is around 200 MB (unpacked).
+This is due to including everything (modules and .NET assemblies) to provide parity with what Windows PowerShell 5.1 with .NET Framework provides.
+There is currently no plan by Windows or the .NET team to ship newer .NET in Windows, so PS7 will need to continue to ship .NET with PS7.
+
+### Support lifecycle
+
+Windows has a 5+5 year support lifecycle.
+.NET supports up to 3 years for Long Term Support (LTS) versions.
+So even if size was not an issue, we cannot check in a version of PS7 with a .NET that is not supported while Windows itself is supported.
+
+### Solution
+
+To solve both of these issues, the proposal is to ship a bootstrapper in Windows that would be supported within the Windows support lifecycle.
+The boostrapper will update PS7 as needed to ensure a supported version of .NET is being used.
+Details of this bootstrapper follow.
+
+## Specification
+
+### pwsh.exe bootstrapper
+
+We would check-in a `pwsh.exe` into Windows 11+ (under `System32`) written in [native code](#native-code) that will:
+
+- Detect if PS7 is already installed
+  - If installed, pass-through all arguments to the new detached `pwsh` child process
+    - This means that if PS7 is already in the `PATH`, then that version (whether installed via MSI, MSIX, or other means)
+      needs to be updated outside of the bootstrapper
+  - Otherwise, [install](#installer) PS7 and then pass-through all arguments to the new detached `pwsh` child process
+
+The boostrapper should explicitly create a detached child `pwsh` process and then exit.
+As a detached child process, it will continue execution even after the parent has exited.
+
+Expectation is that the bootstrapper will always be first in the `$env:PATH` and executed first, so it can also serve
+to [auto-update](#auto-updating) PS7.
+
+This will allow `pwsh -c customer_script.ps1` to work seamlessly as though PS7 was available inbox.
+
+The bootstrapper would need to start the real `pwsh.exe` and bind stdin/stdout so that inputs and outputs come from `pwsh.exe` seamlessly.
+Unlike Unix, there isn't a way to `exec` a new process that replaces itself so for the first execution, there would be two `pwsh.exe` processes.
+
+However, if the bootstrapper detects that stdin, stdout, or stderr is redirected, then it would need to not exit and instead
+continue to redirect until the child `pwsh.exe` process has exited.
+
+Additional Windows requirements:
+
+- registered with Windows so that `Start->Run->pwsh` will start the bootstrapper
+- a separate Windows feature that can be uninstalled (so that using `pwsh` would not automatically work, or is being installed by other means)
+- different icon for the bootstrapper to indicate it is a bootstrapper
+
+### .ps1 registration
```

Comment:
ps1 registration/integration makes sense as feature of PowerShell installer; not something for bootstrapper to be solving.